gradle/build-cache-configuration.settings.gradle.kts

```diff
@@ -0,0 +1,78 @@
+import java.net.URI
+
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This script is applied to the settings in buildSrc and the main build. That is why we
+ * need this to be a script unless we can model dual usage better with composite/included builds or another solution.
+ */
+
+/* Currently IntelliJ assumes that script plugins configure project objects. That is why we need to declare the receiver
+ * object for the functions explicitly to have IDE support. The build would run successfully without declaring
+ * the receiver object, if this script is applied from a settings.gradle.kts
+ *
+ * Note that this script has the ending .settings.gradle.kts. This will eventually used to make IntelliJ aware that
+ * this is configuring a settings object.
+ *
+ */
+
+
+
+val remoteCacheUrl = System.getProperty("gradle.cache.remote.url")?.let { URI(it) }
+val isCiServer = System.getenv().containsKey("CI")
+val remotePush = !(System.getProperty("gradle.cache.remote.push") == "false")
+val remoteCacheUsername = System.getProperty("gradle.cache.remote.username")
+val remoteCachePassword =  System.getProperty("gradle.cache.remote.password")
+
+fun Settings.configureLocalCache() {
+    buildCache {
+        local {
+            isEnabled = !isCiServer
+        }
+    }
+}
+
+fun Settings.configureRemoteCache() {
+    val isRemoteBuildCacheEnabled = remoteCacheUrl != null && gradle.startParameter.isBuildCacheEnabled && !gradle.startParameter.isOffline
+
+    gradle.rootProject {
+        this.extra.set("remoteBuildCacheEnabled", isRemoteBuildCacheEnabled)
```

Comment:
I think we can retire the property. This was used for validation, but all validation moved to `AvailableJavaInstallations.validateBuildCacheConfiguration`.

Reply:
Great. Every extra property we can remove is progress.